toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -245,102 +338,150 @@ func (im *IsoMaker) copyShimFromInitrd(efiBootImgTempMountDir, bootBootloaderFil
+	err = os.MkdirAll(isoRpmsRepoDirPath, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed to mkdir '%s'.", isoRpmsRepoDirPath)
+	}
 
 	fetchedRepoDirContentsPath := filepath.Join(im.fetchedRepoDirPath, "*")
-	recursiveCopyDereferencingLinks(fetchedRepoDirContentsPath, isoRpmsRepoDirPath)
+	err = recursiveCopyDereferencingLinks(fetchedRepoDirContentsPath, isoRpmsRepoDirPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // prepareWorkDirectory makes sure we start with a clean directory
 // under "im.buildDirPath". The work directory will contain the contents of the ISO image.
-func (im *IsoMaker) prepareWorkDirectory() {
+func (im *IsoMaker) prepareWorkDirectory() (err error) {
 	logger.Log.Infof("Building ISO under '%s'.", im.buildDirPath)
 
 	exists, err := file.DirExists(im.buildDirPath)
-	logger.PanicOnError(err, "Failed while checking if directory '%s' exists.", im.buildDirPath)
+	if err != nil {
+		return fmt.Errorf("failed while checking if directory '%s' exists:\n%w", im.buildDirPath, err)
+	}
 	if exists {
 		logger.Log.Warningf("Unexpected: temporary ISO build path '%s' exists. Removing.", im.buildDirPath)
-		logger.PanicOnError(os.RemoveAll(im.buildDirPath), "Failed while removing directory '%s'.", im.buildDirPath)
+		err = os.RemoveAll(im.buildDirPath)
+		if err != nil {
+			return fmt.Errorf("failed while removing directory '%s':\n%w", im.buildDirPath, err)
+		}
 	}
 
-	logger.PanicOnError(os.Mkdir(im.buildDirPath, os.ModePerm), "Failed while creating directory '%s'.", im.buildDirPath)
+	err = os.Mkdir(im.buildDirPath, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed while creating directory '%s':\n%w", im.buildDirPath, err)
+	}
 
-	im.deferIsoMakerCleanUp(func() {
+	im.deferIsoMakerCleanUp(func() error {
 		logger.Log.Debugf("Removing '%s'.", im.buildDirPath)
-
-		logger.PanicOnError(os.RemoveAll(im.buildDirPath), "Failed to remove '%s'.", im.buildDirPath)
+		err := os.RemoveAll(im.buildDirPath)
+		if err != nil {
+			err = fmt.Errorf("failed to remove '%s':\n%w", im.buildDirPath, err)
+		}
+		return err
 	})
 
-	im.copyStaticIsoRootFiles()
+	err = im.copyStaticIsoRootFiles()
+	if err != nil {
+		return err
+	}
 
-	im.copyArchitectureDependentIsoRootFiles()
+	err = im.copyArchitectureDependentIsoRootFiles()
+	if err != nil {
+		return err
+	}
 
-	im.copyAndRenameConfigFiles()
+	err = im.copyAndRenameConfigFiles()
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // copyStaticIsoRootFiles copies architecture-independent files from the
 // Mariner repo directories.
-func (im *IsoMaker) copyStaticIsoRootFiles() {
+func (im *IsoMaker) copyStaticIsoRootFiles() (err error) {
 
 	if im.resourcesDirPath == "" && im.grubCfgPath == "" {
-		logger.Log.Panicf("missing required parameters. Must specify either the resources directory or provide a grub.cfg.")
+		return fmt.Errorf("missing required parameters. Must specify either the resources directory or provide a grub.cfg.")
```

Comment:
I spotted this just now: with `fmt.Errorf` it's a Golang convention we try to follow to not include a period at the end:
```suggestion
		return fmt.Errorf("missing required parameters. Must specify either the resources directory or provide a grub.cfg")
```

Reply:
fixed.